src/Analyser/TypeSpecifier.php

```diff
@@ -2192,14 +2195,11 @@ public function resolveIdentical(Expr\BinaryOp\Identical $expr, Scope $scope, Ty
 			}
 		}
 
-		if ($rightType->isInteger()->yes() || $rightType->isString()->yes()) {
+		if ($rightType->isString()->yes()) {
```

Comment:
as we no longer do `ConstantIntegerType` related checks in `specifyTypesForConstantBinaryExpression` we can simplify this loop again to what it looked like a few weeks ago.